org.chromium.Chromium.yaml

```diff
@@ -0,0 +1,229 @@
+      - --enable-shared
+      - --with-ensurepip=yes
+      - --with-system-expat
+      - --with-system-ffi
+      - --with-dbmliborder=gdbm
+      - --enable-unicode=ucs4
+    build-options:
+      no-debuginfo: true
+    post-install:
+      - chmod 644 $FLATPAK_DEST/lib/libpython2.7.so.1.0
+    cleanup:
+      - '*'
+    sources:
+      - type: archive
+        url: https://www.python.org/ftp/python/2.7.18/Python-2.7.18.tar.xz
+        sha256: b62c0e7937551d0cc02b8fd5cb0f544f9405bafc9a54d3808ed4594812edef43
+
+  - name: xcbproto
+    build-options:
+      env:
+        PYTHON: /app/bin/python2
+    cleanup:
+      - '*'
+    sources:
+      - type: archive
+        url: https://xorg.freedesktop.org/archive/individual/proto/xcb-proto-1.14.tar.xz
+        sha256: 186a3ceb26f9b4a015f5a44dcc814c93033a5fc39684f36f1ecc79834416a605
+
+  - name: krb5
+    subdir: src
+    config-opts:
+      - --disable-static
+      - --disable-rpath
+    sources:
+      - type: archive
+        url: https://kerberos.org/dist/krb5/1.18/krb5-1.18.2.tar.gz
+        sha256: c6e4c9ec1a98141c3f5d66ddf1a135549050c9fab4e9a4620ee9b22085873ae0
+
+  - name: krb5-config
+    buildsystem: simple
+    build-commands:
+      - install -Dm 644 krb5.conf -t /app/etc
+    sources:
+      - type: file
+        path: krb5.conf
+
+  - name: extensions
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/native-messaging-hosts /app/widevine
+
+  - name: chromium
+    buildsystem: simple
+    build-options:
+      env:
+        CC: clang
+        CXX: clang++
+        AR: ar
+        NM: nm
+        CFLAGS: -Wno-unknown-warning-option -Wno-unknown-pragmas
+        CXXFLAGS: -Wno-unknown-warning-option -Wno-unknown-pragmas
+    build-commands:
+      # NOTE: zlib can be unbundled if we added Fedora's unmangling patches
+      # flac can likely be unbundled
+      - >-
+        build/linux/unbundle/replace_gn_files.py --system-libraries
+        libdrm libjpeg libpng libvpx libwebp libxml libxslt opus
+      # Avoid mangling issues
+      - rm third_party/libpng/*.h
+      # DO NOT REUSE THE BELOW API KEYS; these are for Flathub only.
+      # http://lists.debian.org/debian-legal/2013/11/msg00006.html
+      - >-
+        tools/gn/bootstrap/bootstrap.py -v --no-clean --gn-gen-args='
+        use_sysroot=false
+        custom_toolchain="//build/toolchain/linux/unbundle:default"
+        host_toolchain="//build/toolchain/linux/unbundle:default"
+        use_udev=false
+        use_lld=true
+        enable_nacl=false
+        blink_symbol_level=0
+        use_gnome_keyring=false
+        use_pulseaudio=true
+        clang_use_chrome_plugins=false
+        is_official_build=true
+        google_api_key="AIzaSyAL6fqCZVFhA7K_qBvz9GO5Z-V1JBcPO0A"
+        google_default_client_id="62778694883-t4a8bf01lumav756cl6ujhhrqsqselq9.apps.googleusercontent.com"
+        google_default_client_secret="eeHsihEqYjhD2zuIROjerl_7"
+        treat_warnings_as_errors=false
+        '
```

Comment:
adding `symbol_level=0` and `is_debug=0` should help make the build smaller and faster

Reply:
also why isn't `ffmpeg_branding="Chrome"` and `proprietary_codecs=true` there if you want codecs?

Reply:
Hmm, do the debug symbols stick around for some reason? flatpak-builder should strip out the debug symbols into a separate extension, so if e.g. the new portal code crashes or similar, we can use coredumpctl and such to look at the stack traces.

Reply:
@refi64  I have no idea why but on my ungoogled-chromium I am having 2GB binaries. Something very weird is going on :(   

Hopefully it's fixed in my latest commit

Reply:
Yep, it's fixed in the latest commit :D

Reply:
For `proprietary_codecs=true` on Endless we always build chromium itself with it set and also `ffmpeg_branding="Chrome"` but note we also build 2 versions of `ffmpeg.so` (one with `proprietary_codecs=true` and another with it set to `false` and branding=`Chromium`) and on free images we ship the `ffmpeg.so` built with no proprietary codecs. The commit msg [here](https://github.com/endlessm/chromium-browser/commit/e2e1d388a7919e27811f46d52d161ee166bd9eaa) explains a bit the rationale.

See also `debian/rules` under the `debian-master` branch of the endless fork.

Reply:
That said I'm not sure it would be Ok to ship chromium with proprietary codecs enabled or if that could have some licensing issues.